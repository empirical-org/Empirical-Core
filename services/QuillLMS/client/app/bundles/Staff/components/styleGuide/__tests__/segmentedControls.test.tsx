import * as React from "react";
import { render } from "@testing-library/react";

import SegmentedControls from "../segmentedControls";

describe('SegmentedControls', () => {
  test('it should render', () => {
    const { asFragment } = render(<SegmentedControls />);
    expect(asFragment()).toMatchSnapshot();
  })
})
