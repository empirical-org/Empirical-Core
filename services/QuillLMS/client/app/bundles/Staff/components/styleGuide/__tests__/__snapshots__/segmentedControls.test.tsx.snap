// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControls it should render 1`] = `
<DocumentFragment>
  <div
    id="segmented-controls"
  >
    <h2
      class="style-guide-h2"
    >
      Segmented Controls
    </h2>
    <div
      class="element-container"
    >
      <h3>
        Standard Segmented Control
      </h3>
      <pre>
          &lt;SegmentedControl
    activeTab="Option One"
    size="small"
    buttons={[
      {
        label: "Option One",
        onClick: handleTabChange
      },
      {
        label: "Option Two",
        onClick: handleTabChange
      },
      {
        label: "Option Three",
        onClick: handleTabChange
      },
    ]}
  /&gt;
            
      </pre>
      <div
        class="dropdown-container input-container inactive has-text  color-options-dropdown searchable "
        role="button"
        tabindex="0"
      >
        <label>
          Size options
        </label>
        <div
          class="dropdown css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="dropdown__control css-13cymwt-control"
          >
            <div
              class="dropdown__value-container dropdown__value-container--has-value css-1fdsijx-ValueContainer"
            >
              <div
                class="dropdown__single-value css-1dimb5e-singleValue"
              >
                Small
              </div>
              <div
                class="dropdown__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="dropdown__input"
                  id="react-select-2-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="segmented-control-container"
      >
        <button
          class="interactive-wrapper segmented-control-button small left active"
          value="Option One"
        >
          <span>
            Option One
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small "
          value="Option Two"
        >
          <span>
            Option Two
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small right "
          value="Option Three"
        >
          <span>
            Option Three
          </span>
        </button>
      </div>
    </div>
    <div
      class="element-container"
    >
      <h3>
        Segmented Control with icons
      </h3>
      <pre>
          &lt;SegmentedControl
    activeTab="Option One"
    size="small"
    buttons={[
      {
        activeIconSrc: WHITE_STAR_ICON_SRC,
        inactiveIconSrc: BLACK_STAR_ICON_SRC,
        label: "Option One",
        onClick: handleTabChange
      },
      {
        activeIconSrc: WHITE_STAR_ICON_SRC,
        inactiveIconSrc: BLACK_STAR_ICON_SRC,
        label: "Option Two",
        onClick: handleTabChange
      },
      {
        activeIconSrc: WHITE_STAR_ICON_SRC,
        inactiveIconSrc: BLACK_STAR_ICON_SRC,
        label: "Option Three",
        onClick: handleTabChange
      },
      {
        activeIconSrc: WHITE_STAR_ICON_SRC,
        inactiveIconSrc: BLACK_STAR_ICON_SRC,
        label: "Option Four",
        onClick: handleTabChange
      },
      {
        activeIconSrc: WHITE_STAR_ICON_SRC,
        inactiveIconSrc: BLACK_STAR_ICON_SRC,
        label: "Option Five",
        onClick: handleTabChange
      }
    ]}
  /&gt;
            
      </pre>
      <div
        class="segmented-control-container"
      >
        <button
          class="interactive-wrapper segmented-control-button small left active"
          value="Option One"
        >
          <img
            alt=""
            src="https://assets.quill.org/images/icons/xs/star-white.svg"
          />
          <span>
            Option One
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small "
          value="Option Two"
        >
          <img
            alt=""
            src="https://assets.quill.org/images/icons/xs/star.svg"
          />
          <span>
            Option Two
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small "
          value="Option Three"
        >
          <img
            alt=""
            src="https://assets.quill.org/images/icons/xs/star.svg"
          />
          <span>
            Option Three
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small "
          value="Option Four"
        >
          <img
            alt=""
            src="https://assets.quill.org/images/icons/xs/star.svg"
          />
          <span>
            Option Four
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small right "
          value="Option Five"
        >
          <img
            alt=""
            src="https://assets.quill.org/images/icons/xs/star.svg"
          />
          <span>
            Option Five
          </span>
        </button>
      </div>
    </div>
    <div
      class="element-container"
    >
      <h3>
        Segmented Control with disabled option
      </h3>
      <pre>
          &lt;SegmentedControl
    activeTab="Option One"
    size="small"
    buttons={[
      {
        label: "Option One",
        onClick: handleTabChange
      },
      {
        label: "Option Two",
        onClick: handleTabChange
      },
      {
        label: "Option Three",
        onClick: handleTabChange
        disabled: true
      },
    ]}
  /&gt;
            
      </pre>
      <div
        class="segmented-control-container"
      >
        <button
          class="interactive-wrapper segmented-control-button small left active"
          value="Option One"
        >
          <span>
            Option One
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small "
          value="Option Two"
        >
          <span>
            Option Two
          </span>
        </button>
        <button
          class="interactive-wrapper segmented-control-button small right disabled "
          disabled=""
          value="Option Three"
        >
          <span>
            Option Three
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
